src/ngx_http_lua_pipe.h

```diff
@@ -0,0 +1,96 @@
+
+/*
+ * Copyright (C) by OpenResty Inc.
+ */
+
+
+#ifndef _NGX_HTTP_LUA_PIPE_H_INCLUDED_
+#define _NGX_HTTP_LUA_PIPE_H_INCLUDED_
+
+
+#include "ngx_http_lua_common.h"
+
+
+#define NGX_HTTP_LUA_PIPE_ET_ERROR         0x0001
+#define NGX_HTTP_LUA_PIPE_ET_CLOSED        0x0002
+#define NGX_HTTP_LUA_PIPE_ET_SYSCALL       0x0003
+#define NGX_HTTP_LUA_PIPE_ET_NOMEM         0x0004
+#define NGX_HTTP_LUA_PIPE_ET_TIMEOUT       0x0005
```

Comment:
What does `ET` stand for? "error tag"? Does not make sense...

Reply:
I wonder if we could just use an anonymous enum to define these, which can be easier and also cleaner.

Reply:
@agentzh 
The `ET` is an abbreviation for `err_type`, like the `FT` is an abbreviation for `ft_type` in cosocket API implementation.